src/UnlockAccountController.js

```diff
@@ -93,45 +104,50 @@ function (Okta, FormController, Enums, FormType, ValidationUtil, ContactSupport,
               innerTooltip: Okta.loc('account.unlock.email.or.username.tooltip', 'login'),
               icon: 'person-16-gray'
             }
-          })
-        ];
-        if (this.settings.get('features.smsRecovery')) {
-          formChildren.push(FormType.View({View: '\
-            <p class="sms-hint">\
-              {{i18n code="recovery.sms.hint" bundle="login"}}\
-            </p>\
-          '}));
+          }));
+          if (smsEnabled) {
+            formChildren.push(FormType.View({View: '\
+              <p class="sms-hint">\
+                {{i18n code="recovery.sms.hint" bundle="login"}}\
+              </p>\
+            '}));
+            this.$el.addClass('forgot-password-sms-enabled');
+            formChildren.push(this.addRecoveryFactorButton('sms-button', 'account.unlock.sendText',
+              Enums.RECOVERY_FACTOR_TYPE_SMS, form));
+          }
+          if(emailEnabled) {
+            this.$el.addClass('forgot-password-email-enabled');
+            formChildren.push(this.addRecoveryFactorButton('email-button', 'account.unlock.sendEmail',
+              Enums.RECOVERY_FACTOR_TYPE_EMAIL, form));
+          }
         }
 
         return formChildren;
       },
       initialize: function () {
-        var form = this;
 
         this.listenTo(this, 'save', function () {
           this.options.appState.set('username', this.model.get('username'));
+          this.model.save();
         });
 
-        if (this.settings.get('features.smsRecovery')) {
-          this.$el.addClass('forgot-password-sms-enabled');
-          this.addButton({
-            attributes: { 'data-se': 'sms-button'},
-            type: 'button',
-            className: 'button-primary sms-button',
-            text: Okta.loc('account.unlock.sendText', 'login'),
-            action: function () {
-              form.clearErrors();
-              if (this.model.isValid()) {
-                this.model.set('factorType', Enums.RECOVERY_FACTOR_TYPE_SMS);
-                form.trigger('save', this.model);
-              }
-            }
-          }, { prepend: true });
-        }
-
         this.listenTo(this.state, 'contactSupport', function () {
           this.add(ContactSupport, '.o-form-error-container');
         });
+      },
+      addRecoveryFactorButton: function (className, labelCode, factorType, form) {
```

Comment:
same nit as in the other file, since this function no longer adds the button itself, but now it returns a button